### DISCUSSION TITLE
How to retrive training states (epoch, ‘loss_val’, etc..)  from a checkpoint?

### DISCUSSION THREAD

**Discussion by acercyc** [Lightning-AI/pytorch-lightning#13701] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I setup a wandb logger and a checkpoint_callback to save model only when the ‘loss_val’ is minimal. 
Of course, the traning continues and stops after several epochs from the checkpoint with the lowest ‘loss_val’

How can I find out what the epoch, loss_val, etc. values are for that checkpoint?
It is possible to use wandb log and find it. It would be nice if there were a native way to get the state of a particular checkpoint.

**Comment by rohitgr7** [comment]
you can load the checkpoint manually using `torch.load` and check out the epoch key to get the epoch and model_checkpoint state_dict to retrieve the score used for that checkpoint.
